pkg/ipamfetcher/customer.go

```diff
@@ -64,11 +72,50 @@ func (d *Device42CustomerFetcher) FetchCustomers(ctx context.Context) ([]domain.
 	}
 	customers := make([]domain.Customer, 0, len(getCustomersResponse.Customers))
 	for _, customer := range getCustomersResponse.Customers {
+		businessUnit := customer.CustomFields.GetValue("Description")
+		if businessUnit == "" {
+			// fallback to customer name
+			businessUnit = customer.Name
+		}
 		customers = append(customers, domain.Customer{
 			ID:            strconv.Itoa(customer.ID),
-			ResourceOwner: customer.ContactInfo,
-			BusinessUnit:  customer.CustomFields.GetValue("Description"),
+			ResourceOwner: getResourceOwner(customer),
+			BusinessUnit:  businessUnit,
 		})
 	}
 	return customers, nil
 }
+
+func getResourceOwner(customer customer) string {
+
+	/*
+	   Find the best resource owner by searching through the contacts
+	   for non-empty Value for the following Key, in this priority order:
+	   1. Team Lead Contact Email
+	   2. Administrative Contact Email
+	   3. SRE Contact Email
+	   4. Technical Contact Email
+	   5. Fall back to The "contact_info" field directly on the customer record
+	*/
+
+	owner := customer.ContactInfo
+	fromPriority := 5
+	for _, contact := range customer.Contacts {
+		if contact.Type == "Team Lead" && contact.Email != "" && fromPriority > 1 {
+			owner = contact.Email
+			fromPriority = 1
+		} else if contact.Type == "Administrative" && contact.Email != "" && fromPriority > 2 {
+			owner = contact.Email
+			fromPriority = 2
+		} else if contact.Type == "SRE" && contact.Email != "" && fromPriority > 3 {
+			owner = contact.Email
+			fromPriority = 3
+		} else if contact.Type == "Technical" && contact.Email != "" && fromPriority > 4 {
+			owner = contact.Email
+			fromPriority = 4
+		}
+	}
```

Comment:
Question: What would happen if all of these if conditions fail?

Reply:
You'd get empty string back, which is what was already happening anyway if the `customer.ContactInfo` was empty, which was the only field we were looking at prior to this PR.

Reply:
You could do a case/switch with fall-through.  Might be slightly more readable.